odxtools/dtcdop.py

```diff
@@ -201,14 +248,52 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
                 dtc = odxlinks.resolve(dtc_proxy, DiagnosticTroubleCode)
                 self._dtcs.append(dtc)
 
-        linked_dtc_dops = [odxlinks.resolve(x, DtcDop) for x in self.linked_dtc_dop_refs]
-        self._linked_dtc_dops = NamedItemList(linked_dtc_dops)
+        for linked_dtc_dop in self.linked_dtc_dops_raw:
+            linked_dtc_dop._resolve_odxlinks(odxlinks)
 
     def _resolve_snrefs(self, context: SnRefContext) -> None:
+        # hack to avoid initializing the DtcDop object multiple
+        # times. This is required, because the linked DTC DOP feature
+        # requires the "parent" DTC DOPs to be fully initialized but
+        # the standard does not define a formal ordering of DTC DOPs.
+        if hasattr(self, "_linked_dtc_dops"):
+            return
+
         super()._resolve_snrefs(context)
 
         self.compu_method._resolve_snrefs(context)
 
         for dtc_proxy in self.dtcs_raw:
             if isinstance(dtc_proxy, DiagnosticTroubleCode):
                 dtc_proxy._resolve_snrefs(context)
+
+        for linked_dtc_dop in self.linked_dtc_dops_raw:
+            linked_dtc_dop._resolve_snrefs(context)
+
+        # add the inherited DTCs from linked DTC DOPs. Note that this
+        # requires that there are no cycles in the "link-hierarchy"
```

Comment:
required by standard, or current logic?

Reply:
If I didn't miss something, the standard does not mention anything about "linked-dop-ref cycles".

That said, although this is not explicitly mentioned in the text, I got the impression that the standard assumes that all linked DTC DOPs are defined in one of the parent diag layers of the layer which contains the DTC-DOP in question...

Reply:
(non-cyclic link graphs are required to get well-defined results, or at least the standard would need to specify how to handle the corner cases that emerge from them.)

Reply:
okay, [changed]( https://github.com/mercedes-benz/odxtools/pull/360/commits/bf81d92c8830238ac0cc19d2d3d388a9e0669848): I played around with the idea of a separate `_is_initialized` member and I think that the result is not worse than the `hasattr()` route (IMO it is not better either, but ...)